plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -398,16 +398,20 @@ func (c *CiscoTelemetryMDT) handleTelemetry(data []byte) {
 		if msgID := msg.GetSubscriptionIdStr(); msgID != "" {
 			tags["subscription"] = msgID
 		}
-		tags["path"] = msg.GetEncodingPath()
+		encodingPath := msg.GetEncodingPath()
+		tags["path"] = encodingPath
 
 		for _, subfield := range keys.Fields {
 			c.parseKeyField(tags, subfield, "")
 		}
 
-		// Parse values
-		for _, subfield := range content.Fields {
-			c.parseContentField(grouper, subfield, "", msg.EncodingPath, tags, timestamp)
+		if content != nil {
+			// Parse values
+			for _, subfield := range content.Fields {
+				c.parseContentField(grouper, subfield, "", encodingPath, tags, timestamp)
+			}
 		}
```

Comment:
How about
```suggestion
		if content == nil {
			continue
		}

		// Parse values
		for _, subfield := range content.Fields {
			c.parseContentField(grouper, subfield, "", encodingPath, tags, timestamp)
		}
```
as this will save an indentation level and reduces the patch size.

Reply:
Your proposal would lead that the delete field would not be included. I solved it by adding the delete field (if enabled) before parsing the values. See: [faac53a](https://github.com/influxdata/telegraf/pull/12345/commits/faac53a914700b4bfe816acc4babe21a2055a7b4)